boards/posix/nrf52_bsim/irq_handler.c

```diff
@@ -365,29 +365,67 @@ void posix_irq_offload(void (*routine)(void *), void *parameter)
 }
 #endif
 
-/**
- * Replacement for ARMs NVIC_SetPendingIRQ()
- *
- * Sets the interrupt IRQn as pending
- * Note:
- * This will interrupt immediately if the interrupt
- * is not masked and irqs are not locked, and this interrupt is higher
- * priority than a possibly currently running interrupt
+/*
+ *  Replacement for ARMs NVIC functions()
  */
 void NVIC_SetPendingIRQ(IRQn_Type IRQn)
 {
 	hw_irq_ctrl_raise_im_from_sw(IRQn);
 }
 
-/**
- *  Replacement for ARMs NVIC_ClearPendingIRQ()
- *  Clear pending interrupt IRQn
- */
 void NVIC_ClearPendingIRQ(IRQn_Type IRQn)
 {
 	hw_irq_ctrl_clear_irq(IRQn);
 }
 
+void NVIC_DisableIRQ(IRQn_Type IRQn)
+{
+	hw_irq_ctrl_disable_irq(IRQn);
+}
+
+void NVIC_EnableIRQ(IRQn_Type IRQn)
+{
+	hw_irq_ctrl_enable_irq(IRQn);
+}
+
+void NVIC_SetPriority(IRQn_Type IRQn, uint32_t priority)
+{
+	hw_irq_ctrl_prio_set(IRQn, priority);
+}
+
+uint32_t NVIC_GetPriority(IRQn_Type IRQn)
+{
+	return hw_irq_ctrl_get_prio(IRQn);
+}
+
+void sys_arch_reboot(int type)
+{
+	exit(type);
+}
+
+void NVIC_SystemReset(void)
+{
+	sys_arch_reboot(0);
```

Comment:
This one is not provided unless CONFIG_REBOOT is set. So instead would you mind doing posix_print_warning("%s called. Exiting\n",__func__); posix_exit(1); here?
Regargind the placing for this function, it does not really belong in this file. Maybe better to place all these in a new cmsis.c file in this directory? (otherwise I guess placing it here is 2nd best)